### PR TITLE
Add MIPS counter device

### PR DESCRIPTION
**Pull request by kssuraaj28**
Do you wish for something like this? Attempt on #2289

### CODE REVIEW

**Comment by anton-bondarev** [comment]
> Do you wish for something like this? Attempt on #2289

Yes. I do.

As it says on [wiki page](https://github.com/embox/embox/wiki/Time) it increases the time measurement precision. It's great if you  try to make sure about this

**Comment by kssuraaj28** [comment]
I was able to observe increased precision of the `time` cmd (I had to modify the command source a bit to get ns displayed).

File: src/drivers/clock/mips_clk/mips_clk.c [hunk 1]
```diff
@@ -27,6 +27,12 @@ static irq_return_t clock_handler(unsigned int irq_nr, void *dev_id) {
 	return IRQ_HANDLED;
 }
 
+static cycle_t mips_clk_read(struct clock_source *cs) {
+	uint32_t cnt = mips_read_c0_count();
+	return cnt;
+}
+
+
```
Comment (anton-bondarev) [inline src/drivers/clock/mips_clk/mips_clk.c:35]: Please remove doubled empty line

File: src/drivers/clock/mips_clk/mips_clk.c [hunk 2]
```diff
@@ -39,6 +45,10 @@ static struct time_event_device mips_event_device  = {
 	.irq_nr = MIPS_IRQN_TIMER
 };
 
+static struct time_counter_device mips_counter_device  = {
+	.read = mips_clk_read,
+	.cycle_hz = SYS_CLOCK
+};
```
Comment (anton-bondarev) [inline src/drivers/clock/mips_clk/mips_clk.c:51]: please add an empty line

**Comment by anton-bondarev** [comment]
> I was able to observe increased precision of the `time` cmd (I had to modify the command source a bit to get ns displayed).

Great. It's enough for this issue.
Please solve my minor comments to merge the PR

**Comment by kssuraaj28** [comment]
I've pushed the changes

**Review by anton-bondarev — APPROVED** [review]
LGTM

**Comment by anton-bondarev** [comment]
@kssuraaj28 Thank you, I've approved it. And I'm merging it soon.

**Pull request merged**